Adding vision.md and roadmap.md (#82)
princeton-ddss/llm-inference-platform: docs/roadmap.md

# Project Roadmap

A breakdown of the functionality we aim to provide in stages:

> 1. Command-line Deployment (no special infrastructure needed)

Reviewer: ```suggestion
1. Command-line deployment of the LLM
```

> 2. GUI Deployment (integration with RC infrastructure needed)

Reviewer: ```suggestion
2. Deployment of the LLM via web-based graphical user interface (GUI)
```

> 3. "Supported" GUI Interaction

Reviewer: ```suggestion
3. Providing easily accessible graphical user interfaces to interact with the LLM once it is running
```

## Status

|     | Deployment | Interaction |
| --- | ---------- | ----------- |
| CLI | testing    | development |
| GUI | tbd        | not started |

## Command-Line Deployment

**Status: Testing**

We provide a command-line tool for generating SLURM jobs that deploy a
pre-configured LM server on a compute node, with helpful utilities for

> forwarding the LM server and querying its API.

Reviewer: ```suggestion
forwarding the port served by the LM and querying its API.
```

## GUI-based Deployment

**Status: tbd**

We provide a web-based graphical interface (like MyDella) with
textboxes/dropdown menus to choose an open-source LM and deploy it on Della.

The deployment interface can be made with
[Open OnDemand](https://openondemand.org/) or a simple web interface

> ## "Supported" GUI Interaction

Reviewer: ```suggestion
## GUIs to interact with the LM
